static/js/message_report.js

```diff
@@ -0,0 +1,115 @@
+
+    // add the report dialog HTML:
+    //  - append() is used to absolutely minimize the DOM overhead of this rarely used feature,
+    //    i.e. we don't want *any* extra per-message DOM elements for msgs that aren't reported.
+    //  - remove() called in case the user attempts to call report multiple times.
+    const html = render_report_message();
+    $target.find(".reportmsg-box").remove();
+    $target.append(html);
+    $target.find(".reportmsg-box").css("width", width + "px");
+
+    // nice little animation to gently open the dialog box and then scroll down to it,
+    // drawing the user's eyes to the reporting form.  note: for uncondensed long messages,
+    // this initial animation (N msec) will happen off-screen, which is fine in practice.
+    const $reportmsg = $row.find(".reportmsg-box");
+    $reportmsg.slideDown(600, () => {
+        message_viewport.$message_pane.animate({scrollTop: scrollDest}, 600);
+    });
+    $reportmsg.find("div,label,input[type=radio]").on("click", (e) => {
+        // don't initiate a reply-message, which is the default behavior in zulip
+        e.stopPropagation();
+    });
+    $reportmsg.find(".reportmsg_cancel").on("click", (e) => {
+        $reportmsg.slideUp(400);
+        e.stopPropagation();
+        e.preventDefault();
+    });
+    $reportmsg.find(".reportmsg_submit").on("click", (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+
+        const reason = $reportmsg.find("input[type=radio][name=reportmsg-reason]:checked").val();
+        if (!["spam", "harassment", "inappropriate", "norms", "other"].includes(reason)) {
+            msgreport_alert($t({defaultMessage: "Please select a reason."}));
+            return;
+        }
+        let explanation = $reportmsg.find("textarea[name=reportmsg-explain]").val();
+        if (explanation === undefined) {
+            explanation = "";
+        }
+        if (reason === "other" && explanation === "") {
+            msgreport_alert($t({defaultMessage: "'Other' requires an explanation."}));
+            return;
+        }
+
+        if (message && message.raw_content) {
+            report_message(message, $reportmsg, reason, explanation);
+            return;
+        }
+        channel.get({
+            url: "/json/messages/" + message.id,
+            idempotent: true,
+            success(data) {
+                message.raw_content = data.raw_content;
+                report_message(message, $reportmsg, reason, explanation);
+            },
+        });
+    });
+}
+
+export function report_message(message, $reportmsg, reason, explanation) {
+    channel.post({
+        url: "/json/messages/report/" + message.id,
+        data: {
+            reason,
+            explanation,
+            quoted_message: compose_actions.get_quoted_message(message),
+        },
+        success() {
+            $reportmsg.html(
+                $t({defaultMessage: "Thanks! Submitted to site administrators for review."}),
```

Comment:
Maybe "Submitted to the organization's moderation team to review."? 

(I think we need to be explicit that it's members of this organization, not the people who run the server, who get the report).

Reply:
DONE.